### PR TITLE
Prepare branch 7.0, Switch license to Apache License Version 2.0 😊🎉

### PR DESCRIPTION
**Pull request by Sanne**
includes:
 - https://hibernate.atlassian.net/browse/HCANN-133
 - https://hibernate.atlassian.net/browse/HCANN-134

### CODE REVIEW

File: src/main/java/org/hibernate/annotations/common/reflection/java/generics/ParameterizedTypeImpl.java [hunk 1]
```diff
@@ -1,10 +1,8 @@
-/*
- * Hibernate, Relational Persistence for Idiomatic Java
+/* Hibernate, Relational Persistence for Idiomatic Java
```
Comment (yrodiere) [inline src/main/java/org/hibernate/annotations/common/reflection/java/generics/ParameterizedTypeImpl.java:1]: Can we consider dropping this tagline?

I don't see the point of it, and it forces us to configure IDEs differently for each Hibernate project, which is annoying.

**Review by yrodiere — APPROVED** [review]
LGTM, though I'd drop the tagline :)

File: src/main/java/org/hibernate/annotations/common/reflection/java/generics/ParameterizedTypeImpl.java (hunk 1 above)
Comment (Sanne) [reply src/main/java/org/hibernate/annotations/common/reflection/java/generics/ParameterizedTypeImpl.java:1]: Ok, right I copied the ASL header from Hibernate Reactive.

How would you have it?

Comment (yrodiere) [reply src/main/java/org/hibernate/annotations/common/reflection/java/generics/ParameterizedTypeImpl.java:1]:
```
/*
 * SPDX-License-Identifier: Apache-2.0
 * Copyright: Red Hat Inc. and Hibernate Authors
 */
```

Or (even shorter):

```
// SPDX-License-Identifier: Apache-2.0
// Copyright: Red Hat Inc. and Hibernate Authors
```

Or whatever @sebersole wants to do on ORM. I'm more concerned about using the same header on all projects than about the exact content of the header, TBH.

Comment (Sanne) [reply src/main/java/org/hibernate/annotations/common/reflection/java/generics/ParameterizedTypeImpl.java:1]: Ok, sure we can do that.

What do other things, let's have a minimalistic header?

Incidentally, headers aren't even mandatory when choosing ASL2 - just a minor preference to make sure contributors are reminded about it once again - but not being essential, if people feel strongly we'd want to be minimalistic, we can also remove them altogether I believe. (I'd need to double check as opinions vary over time.. it was like this last time we discussed it.)

Comment (Sanne) [reply src/main/java/org/hibernate/annotations/common/reflection/java/generics/ParameterizedTypeImpl.java]: Resolved as per decision on https://hibernate.zulipchat.com/#narrow/stream/132094-hibernate-orm-dev/topic/Preferred.20header

**Pull request merged**